### PR TITLE
[ISSUE #5823] Add delete topic in message store interface

### PR DESCRIPTION
**Pull request by lizhimins**

### CODE REVIEW

File: broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java [hunk 1]
```diff
@@ -1960,11 +1960,20 @@ public RemotingCommand deleteExpiredCommitLog() {
...
-        LOGGER.warn("invoke cleanUnusedTopic start.");
+        LOGGER.info("AdminBrokerProcessor#cleanUnusedTopic: start.");
         final RemotingCommand response = RemotingCommand.createResponseCommand(null);
-        brokerController.getMessageStore().cleanUnusedTopic(brokerController.getTopicConfigManager().getTopicConfigTable().keySet());
-        LOGGER.warn("invoke cleanUnusedTopic end.");
-        response.setCode(ResponseCode.SUCCESS);
+        Set<String> topicManagerTopics = Sets.newHashSet(
+            brokerController.getTopicConfigManager().getTopicConfigTable().keySet());
+        int responseCode = ResponseCode.SUCCESS;
+        if (this.brokerController.getMessageStore() instanceof DefaultMessageStore) {
```
Comment (ShadowySpirits) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java:1968]: Why should we add this restriction？ IMO, there is no need to care about the underlying implementation outside the store module.

**Review by RongtongJin — COMMENTED** [review]
We have deleteTopics and cleanUnusedTopic API, I don't think there is anything wrong with the semantics of cleanUnusedTopic. Or we can use deleteTopic instead of cleanUnusedTopic in message store.

File: broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java (hunk 1 above)
Comment (RongtongJin) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java:1968]: I guess only the implementation class DefaultMessageStore can obtain consumeQueueTable

**Comment by lizhimins** [comment]
image

"delete topic" is a remoting API，"clean unused topic" is a store API

**Comment by RongtongJin** [comment]
> image
> 
> "delete topic" is a remoting API，"clean unused topic" is a store API

“clean unused topic” is also a remoting API, but it has different meanings from the store API "clean unused topic", so why not rename store API "clean unused topic" to "delete topics" ?

File: broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java (hunk 1 above)
Comment (ShadowySpirits) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java:1968]: Then we maybe consider exposing the interface to get all cq or sinking the implementation to the store module.

Comment (lizhimins) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java]: I already change it.

**Review by ShadowySpirits — APPROVED** [review]

**Review by aaron-ai — APPROVED** [review]

**Pull request merged**